pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,275 @@
+    color_light = "white"
+
+    # visual
+    color_blue = "48/105/152"  # Python blue
+    color_yellow = "255/212/59"  # Python yellow
+    color_red = "238/86/52"  # GMT red
+    if blackwhite:
+        color_blue = color_yellow = color_red = color_light
+        if not darkmode:
+            color_blue = color_yellow = color_red = color_dark
+
+    # background and wordmark
+    color_bg = color_dark
+    color_py = color_yellow
+    color_gmt = color_light
+    if not darkmode:
+        color_bg = color_light
+        color_py = color_blue
+        color_gmt = color_dark
+
+    # -----------------------------------------------------------------------------
+    # Define shape
+    # -----------------------------------------------------------------------------
+    symbol = "c"  # circle
+    diameter = 7.5
+    diameter_add = 0.5
+    if hexshape:
+        symbol = "h"  # hexagon
+        diameter = 8.6
+        diameter_add = 0.6
+
+    # -----------------------------------------------------------------------------
+    # Define wordmark
+    # -----------------------------------------------------------------------------
+    font = "AvantGarde-Book"
+    match wordmark:
+        case "vertical":
+            args_text_wm = {
+                "x": 0,
+                "y": -5,
+                "justify": "CT",
+                "font": f"2.5c,{font}",
+            }
+        case True | "horizontal":
+            args_text_wm = {
+                "x": 6,
+                "y": 0,
+                "justify": "LM",
+                "font": f"8c,{font}",
+            }
+
+    # -----------------------------------------------------------------------------
+    # Start plotting
+    # -----------------------------------------------------------------------------
+    fig = pygmt.Figure()
+
+    # .............................................................................
+    # blue circle / hexagon for Earth
+    # .............................................................................
+    fig.plot(
+        x=0,
+        y=0,
+        region=region,
+        projection=f"X{size * 2}c",
+        style=f"{symbol}{diameter}c",
+        pen=f"15p,{color_blue}",
+        fill=color_bg,
+        no_clip=True,
+        perspective=perspective,
+    )
+
+    # .............................................................................
+    # yellow lines for compass
+    # .............................................................................
+    args_yellow = {"pen": f"5p,{color_yellow}", "perspective": True}
+    # horizontal yellow line
+    fig.plot(x=[-4, 4], y=[0, 0], no_clip=True, **args_yellow)
+    # diagonal yellow lines
+    lines_diagonal = [
+        ([-r1, -r2], [r1, r2]),  # upper left
+        ([-r1, -r2], [-r1, -r2]),  # lower left
+        ([r1, r2], [r1, r2]),  # lower left
+        ([r1, r2], [-r1, -r2]),  # lower right
+    ]
+    for x, y in lines_diagonal:
+        fig.plot(x=x, y=y, **args_yellow)
```

Comment:
`no_clip` has no effects here, right? If yes, then it can be simplified to:
```suggestion
    lines = [
        ([-size, size], [0, 0]),   # horizontal line
        ([-r1, -r2], [r1, r2]),  # upper left
        ([-r1, -r2], [-r1, -r2]),  # lower left
        ([r1, r2], [r1, r2]),  # lower left
        ([r1, r2], [-r1, -r2]),  # lower right
    ]
    for x, y in lines:
        fig.plot(x=x, y=y, pen=f"5p,{color_yellow}", perspective=True})
```